craft_application/models/project.py

```diff
@@ -118,21 +119,93 @@ class Project(base.CraftBaseModel):
+    describing what the project does and who may find it useful.
+    """
+
+    base: str | None
+    build_base: str | None
+
+    platforms: PlatformsDict = pydantic.Field(
+        description="Determines which architectures the project builds and runs on.",
+        examples=[
+            "{amd64: {build-on: [amd64], build-for: [amd64]}, arm64: {build-on: [amd64, arm64], build-for: [arm64]}}"
+        ],
+    )
+    """Determines which architectures the project builds and runs on.
+
+    If the platform name is a valid Debian architecture, the ``build-on`` and
+    ``build-for`` keys can be omitted.
+
+    The platform name describes a ``build-on``-``build-for`` pairing.  When setting
+    ``build-on`` and ``build-for``, the name is arbitrary but it's recommended to match
+    the platform name to the architecture set by ``build-for``.
+    """
+
+    contact: str | UniqueStrList | None = pydantic.Field(
+        default=None,
+        description="The author's contact links and email addresses.",
+        examples=["[contact@example.com, https://example.com/contact]"],
+    )
+
+    issues: str | UniqueStrList | None = pydantic.Field(
+        default=None,
+        description="The links and email addresses for submitting issues, bugs, and feature requests.",
+        examples=["[issues@example.com, https://example.com/issues]"],
+    )
+
+    source_code: pydantic.AnyUrl | None = pydantic.Field(
+        default=None,
+        description="The links to the source code of the project.",
+        examples=["[https://github.com/canonical/craft-application]"],
+    )
+
+    license: str | None = pydantic.Field(
+        default=None,
+        description="The project's license as an SPDX expression",
+        examples=["GPL-3.0+", "Apache-2.0"],
+    )
+    """Currently, only `SPDX 2.1 expressions <https://spdx.org/licenses/>`_ are
+    supported.
+
+    For “or later” and “with exception” license styles, refer to `Appendix V of the SPDX
+    Specification 2.1
+    <https://web.archive.org/web/20230902152422/https://spdx.dev/spdx-specification-21-web-version/#h.twlc0ztnng3b>`_.
+    """
 
-    adopt_info: str | None = None
+    adopt_info: str | None = pydantic.Field(
+        default=None,
+        description="Selects a part to inherit metadata from.",
+        examples=["foo-part"],
+    )
 
     # parts are handled by craft-parts
-    parts: dict[str, Part]
+    parts: dict[str, Part] = pydantic.Field(
+        description="The self-contained software pieces needed to create the final artifact.",
+        examples=[
+            textwrap.dedent(
+                "{cloud-init: \
+                {plugin: python, \
+                source-type: git, \
+                source: https://git.launchpad.net/cloud-init}}"
+            ),
+        ],
+    )
+    """The self-contained software pieces needed to create the final artifact.
+
+    Each part is pulled, built, staged, and primed according to the parts lifecycle
```

Comment:
```suggestion
    Each part is pulled, overlaid, built, staged, and primed according to the parts lifecycle
```

(I am subtly lobbying for the OVERLAY step :D)

Reply:
I've removed the sentence to make the description more app-agnostic. On second glance, I didn't find it useful without a link, and I'm trying to avoid directing users to different RTD projects.